test/e2e/app-dir/back-forward-cache/back-forward-cache.test.ts

```diff
@@ -174,27 +180,37 @@ describe('back/forward cache', () => {
     // Navigate to 2 additional pages.
     const linkToPage3 = await browser.elementByCss('a[href="/page/3"]')
     await linkToPage3.click()
+    await retry(async () => {
+      expect(await browser.elementByCss('h2').text()).toEqual('Page 3')
+    })
     const linkToPage4 = await browser.elementByCss('a[href="/page/4"]')
     await linkToPage4.click()
+    await retry(async () => {
+      expect(await browser.elementByCss('h2').text()).toEqual('Page 4')
+    })
```

Comment:
This is the important bit. I suspect we sometimes never actually finished navigating to page 4 and then the bfcache for page 2 was still active when we navigated back.